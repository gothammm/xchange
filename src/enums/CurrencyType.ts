enum CurrencyType {
  USD = "USD",
  EUR = "EUR",
  GBP = "GBP"
}

export default CurrencyType;