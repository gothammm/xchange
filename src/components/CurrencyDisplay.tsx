import React from 'react';
import { Typography } from 'antd';
import CurrencyType from '../enums/CurrencyType';

interface CurrencyDisplayProps {
  value: number;
  currency: CurrencyType;
}

const getDisplayString = (value: number, currency: CurrencyType) =>
  value.toLocaleString('en-US', {
    style: 'currency',
    currency,
  });

const CurrencyDisplay: React.FC<CurrencyDisplayProps> = ({
  currency,
  value,
}) => {
  return (
    <Typography.Title>{getDisplayString(value, currency)}</Typography.Title>
  );
};

export default CurrencyDisplay;
