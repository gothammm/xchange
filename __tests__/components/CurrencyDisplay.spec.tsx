import React from 'react';
import renderer from 'react-test-renderer';
import CurrencyDisplay from '../../src/components/CurrencyDisplay';
import CurrencyType from '../../src/enums/CurrencyType';

describe(`CurrencyDisplay component`, () => {
  test(`should match snapshot for various currency type`, () => {
    const usdCurrencyComponent = renderer.create(
      <CurrencyDisplay value={60} currency={CurrencyType.USD}></CurrencyDisplay>
    );
    expect(usdCurrencyComponent.toJSON()).toMatchSnapshot();
    const eurCurrencyComponent = renderer.create(
      <CurrencyDisplay value={60} currency={CurrencyType.EUR}></CurrencyDisplay>
    );
    expect(eurCurrencyComponent.toJSON()).toMatchSnapshot();
    const gbpCurrencyComponent = renderer.create(
      <CurrencyDisplay value={60} currency={CurrencyType.GBP}></CurrencyDisplay>
    );
    expect(gbpCurrencyComponent.toJSON()).toMatchSnapshot();
  });
});
