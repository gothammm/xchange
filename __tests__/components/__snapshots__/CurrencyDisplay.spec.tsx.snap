// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencyDisplay component should match snapshot for various currency type 1`] = `
<h1
  className="ant-typography"
  style={
    Object {
      "WebkitLineClamp": null,
    }
  }
>
  $60.00
</h1>
`;

exports[`CurrencyDisplay component should match snapshot for various currency type 2`] = `
<h1
  className="ant-typography"
  style={
    Object {
      "WebkitLineClamp": null,
    }
  }
>
  €60.00
</h1>
`;

exports[`CurrencyDisplay component should match snapshot for various currency type 3`] = `
<h1
  className="ant-typography"
  style={
    Object {
      "WebkitLineClamp": null,
    }
  }
>
  £60.00
</h1>
`;
